packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -146,17 +148,19 @@ export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
 
     const currentId = id.current;
     setIsCreatingSlices(true);
-    addSlices(newSlices)
-      .then(() => {
-        if (currentId !== id.current) return;
-        setIsCreatingSlices(false);
-        // TODO: Execute modal callback.
-      })
-      .catch(() => {
-        if (currentId !== id.current) return;
-        setIsCreatingSlices(false);
-        // TODO: Show error?
-      });
+
+    try {
+      const slicesAdded = await addSlices(newSlices);
+      if (currentId !== id.current) return;
+      await onSuccess(slicesAdded);
```

Comment:
we shouldn't await here - if for some reason `onSuccess` fails, the catch would continue but in theory the modal has finished its job already

Reply:
Done ☑️